core/sql/executor/ExHbaseIUD.cpp

```diff
@@ -924,6 +938,58 @@ ExWorkProcRetcode ExHbaseAccessUpsertVsbbSQTcb::work()
 	  }
 	  break;
 
+        case RETURN_ROW:
+          {
+            if (qparent_.up->isFull())
+	      return WORK_OK;
+	   
+	    if (returnUpdateExpr())
+	      {
+		ex_queue_entry * up_entry = qparent_.up->getTailEntry();
+
+	 	// allocate tupps where returned rows will be created
+                if (allocateUpEntryTupps(
+                         -1,
+                         0,
+                         hbaseAccessTdb().returnedTuppIndex_,
+                         hbaseAccessTdb().returnUpdatedRowLen_,
+                         FALSE,
+                         &rc))
+              return rc;
+
+                ex_expr::exp_return_type exprRetCode =
+		  returnUpdateExpr()->eval(up_entry->getAtp(), workAtp_);
+		if (exprRetCode == ex_expr::EXPR_ERROR)
+		  {
+		    step_ = HANDLE_ERROR;
+		    break;
+		  }
+		
+		rc = 0;
+		// moveRowToUpQueue also increments matches_
+		if (moveRowToUpQueue(&rc))
+		  return 1;
```

Comment:
This was cloned from other parts of this file. I have changed it in this case to return "rc". That's what is supposed to be returned.